bundles/wps/wps-service/src/test/java/org/orbisgis/wpsservice/controller/execution/DataProcessingTest.java

```diff
@@ -0,0 +1,262 @@
+     * @throws MalformedScriptException
+     */
+    @Test
+    public void testGeometryProcessing() throws MalformedScriptException {
+        //Store into the dataMap the geometryData data
+        URI uri = URI.create(UUID.randomUUID().toString());
+        Map<URI, Object> dataMap = new HashMap<>();
+        dataMap.put(uri, "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))");
+        //Generate the InputDescriptionType object containing the GeometryData
+        List<Format> formatList = FormatFactory.getFormatsFromExtensions(new String[]{".txt"});
+        List<DataType> geometryList = new ArrayList<>();
+        geometryList.add(DataType.POLYGON);
+        GeometryData geometryData = new GeometryData(formatList, geometryList);
+        geometryData.setExcludedTypeList(new ArrayList<DataType>());
+        geometryData.setDimension(2);
+        CodeType identifier = new CodeType();
+        identifier.setValue(uri.toString());
+        InputDescriptionType input = new InputDescriptionType();
+        input.setIdentifier(identifier);
+        input.setDataDescription(objectFactory.createGeometryData(geometryData));
+        //Preprocess the GeometryData
+        dataProcessingManager.preProcessData(input, dataMap, null);
+        //Test if the result object is the expected geometry
+        Object result = dataMap.get(uri);
+        Assert.assertTrue("The object resulting of the preprocessing of the geometry data should be a geometry.",
+                result instanceof Geometry);
+        Geometry geometry = (Geometry)result;
+        Assert.assertEquals("The geometry dimension should be 2.",
+                2, geometry.getDimension());
+        Assert.assertEquals("The bounding box geometry wasn't the one expected.",
+                "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", wktWriter.write(geometry));
+
+        //Generate the OutputDescriptionType object to postprocess
+        OutputDescriptionType output = new OutputDescriptionType();
+        output.setIdentifier(identifier);
+        output.setDataDescription(objectFactory.createGeometryData(geometryData));
+        //Postprocess the output
+        dataProcessingManager.postProcessData(output, dataMap, new HashMap<URI, Object>() ,null);
+        //Test if the result object is a string containing the representation of the GeometryData
+        Object result2 = dataMap.get(uri);
+        Assert.assertTrue("The object resulting of the postprocessing of the bounding box should be a String.",
+                result2 instanceof String);
+        String str = result2.toString();
+        Assert.assertEquals("The bounding box geometry wasn't the one expected.",
+                "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", str);
+    }
+
+    /**
+     * Test the preprocessing in the different cases of wrong GeometryData configuration (bad geometry, bad dimension ...)
+     * @throws MalformedScriptException
+     */
+    @Test
+    public void testBadGeometryProcessing() throws MalformedScriptException {
```

Comment:
I think you must add you unit test to follow this practice 

@Test(expected = MalformedScriptException.class)
 public void testBadGeometryProcessing() throws Throwable {

see here an example https://github.com/orbisgis/h2gis/blob/4ba5249a4a229c38d30323a152c1d8d5524fc776/h2gis-functions/src/test/java/org/h2gis/functions/spatial/SpatialFunctionTest.java#L176